x/stakeibc/keeper/icqcallbacks_validator_exchange_rate.go

```diff
@@ -84,15 +83,13 @@ func ValidatorExchangeRateCallback(k Keeper, ctx sdk.Context, args []byte, query
 	hostZone.Validators[valIndex] = &validator
 	k.SetHostZone(ctx, hostZone)
 
-	k.Logger(ctx).Info(fmt.Sprintf("ValidatorCallback: HostZone %s, Validator %v, tokensFromShares %v",
-		hostZone.ChainId, validator.Address, validator.InternalExchangeRate.InternalTokensToSharesRate))
+	k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_Validator, "Validator Internal Exchange Rate: %v",
+		validator.InternalExchangeRate.InternalTokensToSharesRate))
 
-	// armed with the exch rate, we can now query the (val,del) delegation
-	err = k.QueryDelegationsIcq(ctx, hostZone, queriedValidator.OperatorAddress)
-	if err != nil {
-		errMsg := fmt.Sprintf("ValidatorCallback: failed to query delegation, zone %s, err: %s", hostZone.ChainId, err.Error())
-		k.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(types.ErrICQFailed, errMsg)
+	// Armed with the exch rate, we can now query the (validator,delegator) delegation
+	if err := k.QueryDelegationsIcq(ctx, hostZone, queriedValidator.OperatorAddress); err != nil {
```

Comment:
what is this syntax? isn't `err != nil` redundant?

```
if err := fn(); err != nil {
   // log or return error
} 
```

Reply:
```go
// this:
if err := fn(); err != nil {
   // log or return error
} 

// is just a shorthand way of doing this:

err := fn()
if err != nil {
    // log or return error
}
```
https://www.calhoun.io/one-liner-if-statements-with-errors/

Is that what you were asking or did I misinterpret your question?

Reply:
Makes sense, thanks. For some reason I hadn't used this syntax before.